utils.go

```diff
@@ -51,15 +76,15 @@ func JoinTraceFromHeader(
 			if err != nil {
 				return nil, err
 			}
-			contextIDMap[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
+			carrier.TracerState[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
```

Comment:
`Open-Tracing-Context-Id-` should definitely go away, we even have a ticket for it.